ts/features/lollipop/utils/__tests__/lollipopRequestInit.test.ts

```diff
@@ -0,0 +1,164 @@
+import { PublicKey } from "@pagopa/io-react-native-crypto";
+import { LollipopConfig } from "../..";
+import { KeyInfo } from "../crypto";
+import { lollipopRequestInit } from "../fetch";
+import { getUnixTimestamp } from "../../httpSignature/signature";
+
+const publicKey: PublicKey = {
+  crv: "P-256",
+  x: "dyLTwacs5ej/nnXIvCMexUBkmdh6ArJ4GPKjHob61mE=",
+  kty: "EC",
+  y: "Tz0xNv++cOeLVapU/BhBS0FJydIcNcV25/ALb1HVu+s="
+};
+
+const publicKeyThumbprint = "SXn6l6BNlwAb60cJUKpvKB3H-UQbe2slQ_8LBR70cfA";
+
+const keyInfo: KeyInfo = {
+  keyTag: "a12e9221-c056-4bbc-8623-ca92df29361e",
+  publicKey,
+  publicKeyThumbprint
+};
+
+const lollipopConfig: LollipopConfig = {
+  nonce: "nonce"
+};
+const fullUrl = "https://localhost:3000/method";
+const requestInit: RequestInit = {
+  headers: {
+    Authorization: "Bearer 123"
+  },
+  method: "GET"
+};
+
+const configurationError =
+  "Bad input parameters, unable to compose RequestAndKeyInfoForLPFetch";
+
+jest.mock("@pagopa/io-react-native-crypto", () => ({
+  sign: jest.fn().mockResolvedValue("MockSignature")
+}));
+
+const testInit = (timestamp: number, signBody: boolean = false) => ({
+  headers: {
+    Authorization: "Bearer 123",
+    "x-pagopa-lollipop-original-method": "GET",
+    "x-pagopa-lollipop-original-url": "https://localhost:3000/method",
+    signature: "sig1=:MockSignature:",
+    "signature-input": `sig1=(${
+      signBody ? '"content-digest" ' : ""
+    }"x-pagopa-lollipop-original-method" "x-pagopa-lollipop-original-url");created=${timestamp};nonce="nonce";alg="ecdsa-p256-sha256";keyid="SXn6l6BNlwAb60cJUKpvKB3H-UQbe2slQ_8LBR70cfA"`
+  },
+  method: "GET"
+});
+
+const testContentDigest =
+  "sha-256=:Iw2DWNyOiJC0xY3utikS7i8gNXrpKlzIYbmOaP4xrLU=:";
+
```

Comment:
```suggestion

MockDate.set("2023-01-01T01:00:00");
const mockTimestamp = 1672534800;

describe("Check test mocks", () => {
  describe("timestamp from getUnixTimestamp()", () => {
    it(`should be ${mockTimestamp}`, () => {
      expect(getUnixTimestamp()).toBe(mockTimestamp);
    });
  });
});

```

To be more "time agnostic" against the `getUnixTimestamp` we could mock the test date. What do you think_

Reply:
It is a useful test but, IMHO, it does not belong here. Where would it be a good place to ~~place~~ put it?

Reply:
It's more a sanity check for the mock setup. Anyway for the "time agnostic" part, only line 56 it's enough.

Reply:
Done it!